main.py

```
@@ -396,71 +389,20 @@ def submit_feedback(req: FeedbackRequest, request: Request):
-        )
-
-    return {"url": url, "title": title[:200], "text": text[:500_000]}
+        src = source_intake.from_url(req.url)
+    except SourceIntakeError as exc:
+        logger.info("intake_failed", extra={
+            "exc_type": type(exc).__name__,
+            "reason": getattr(exc, "reason", None),
+            "cause": getattr(exc, "cause", None),
+            "url_summary": _summarize_url_for_log(req.url),
+        })
+        raise _map_intake_error(exc)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)